lib/inspector.js

```diff
@@ -188,11 +189,26 @@ function inspectorWaitForDebugger() {
     throw new ERR_INSPECTOR_NOT_ACTIVE();
 }
 
+function emit(eventName, params) {
```

Comment:
Is there a reason why this cannot be done as part of  `Session#post` though? This wouldn't work well in the case that an error is occurred - the callback-based `Session#post` API is also problematic but the promisified API works pretty well.

Reply:
I think for the internal API, it's fine to invent an internal helper when we are 100% sure that the emitted event won't fail. For the public API, it should be delegated to the (promisified) `Session#post` as much as possible, because otherwise errors can be swallowed.

Reply:
An inspector frontend (like ChromeDevTools or `inspector.Session`) sends protocol commands (e.g. `Runtime.enable`) and receives responses and errors. In contrast, an inspector backend (a Node.js app with the inspector enabled) sends protocol events to all connected devtools frontend instances. I think `Session#post` can't be applied for protocol events and we don't have to handle errors in this context.

Reply:
Ah right, this is server -> client. In that case I think this should be renamed to something like `broadcastToFrontend` to disambiguate (this also doesn't really allow communication with a specific frontend, and sends the event to all channels).